pkg/aws/core_util.go

```diff
@@ -265,16 +266,17 @@ func terminateInstance(req *driver.DeleteMachineRequest, svc ec2iface.EC2API, ma
 	_, err := svc.TerminateInstances(input)
 	if err != nil {
 		// if InvalidInstanceID.NotFound error from AWS, then assume VM is terminated
-		if strings.Contains(err.Error(), ec2.UnsuccessfulInstanceCreditSpecificationErrorCodeInvalidInstanceIdNotFound) {
+		errcode := awserror.GetMCMErrorCodeForTerminateInstances(err)
+		if errcode == codes.NotFound {
+			klog.V(2).Infof("no backing VM for %s machine found", req.Machine.Name)
 			return nil
```

Comment:
we wanted to send the `NotFound` error as return right, or am I forgetting some discussion..

Reply:
MCM code will have to be changed if I return `NotFound` error code.